pkg/k8s/node.go

```diff
@@ -40,6 +41,10 @@ func UncordonNode(name string, client kubernetes.Interface) error {
 func IsCordoned(name string, client kubernetes.Interface) (bool, error) {
 	node, err := client.CoreV1().Nodes().Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil {
+		// considered node is cordoned when it's removed from kube api
+		if apierrors.IsNotFound(err) {
```

Comment:
Imo it should be up to the called to decide what to do with the error, not the function itself. When handling the error returned before making pre-termination checks, that is where we can parse the error that has been returned.

Reply:
I see it more as errors needing to be handled at the top-level rather than in functions. This function is just meant to be a wrapper for making the api call and shouldn't be opinionated about the error returned, that should be up to the caller. A different caller may want to handle the error differently, and adding this functionality to the function itself may block that separate use-case.

Reply:
By putting it lower in the stack the method itself becomes rigid and catering to the existing use-case. What if in the future the function can be used but because it does include that error skipping it cannot be used. The way I see it, the caller is the one that can meaningfully handle the error, not the function itself. Right now the function is simply ignoring the error, which is what the caller would be doing in this case. You're essentially putting in functionality from the caller in the function.